src/field/field255.rs

```diff
@@ -0,0 +1,483 @@
+    field::{FieldElement, FieldElementVisitor, FieldError},
+};
+use fiat_crypto::curve25519_64::{
+    fiat_25519_add, fiat_25519_carry, fiat_25519_carry_mul, fiat_25519_from_bytes, fiat_25519_opp,
+    fiat_25519_sub, fiat_25519_tight_field_element, fiat_25519_to_bytes,
+};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    convert::TryFrom,
+    fmt::{self, Debug, Display, Formatter},
+    io::{Cursor, Read},
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+};
+use subtle::{Choice, ConstantTimeEq, ConstantTimeGreater, ConstantTimeLess};
+
+#[cfg(test)]
+pub(crate) mod integer;
+
+#[cfg(test)]
+use integer::Field255Integer;
+
+// `python3 -c "print(', '.join(hex(x) for x in (2**255-19).to_bytes(32, 'big')))"`
+const MODULUS_BIG_ENDIAN: [u8; 32] = [
+    0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xed,
+];
+
+/// `GF(2^255 - 19)`, a 255-bit field.
+#[derive(Clone, Copy, Default)]
+pub struct Field255(fiat_25519_tight_field_element);
+
+impl Field255 {
+    /// Attempts to instantiate a `Field255` from the first `Self::ENCODED_SIZE` bytes in the
+    /// provided slice.
+    ///
+    /// # Errors
+    ///
+    /// An error is returned if the provided slice is not long enough to encode a field element or
+    /// if the decoded value is greater than the field prime.
+    fn try_from_bytes(bytes: &[u8], mask_top_bit: bool) -> Result<Self, FieldError> {
+        if Self::ENCODED_SIZE > bytes.len() {
+            return Err(FieldError::ShortRead);
+        }
+
+        let mut value = [0u8; Self::ENCODED_SIZE];
+        value.copy_from_slice(&bytes[..Self::ENCODED_SIZE]);
+
+        if mask_top_bit {
+            value[0] &= 0b0111_1111;
+        }
+
+        // Walk through the big-endian encoding of the provided value, and identify whether the
+        // first byte that differs from the field's modulus is less than the corresponding byte or
+        // greater than the corresponding byte, in constant time. (or whether the provided value is
+        // equal to the field modulus)
+        let mut less_than_modulus = Choice::from(0u8);
+        let mut greater_than_modulus = Choice::from(0u8);
+        for (value_byte, modulus_byte) in value.iter().zip(MODULUS_BIG_ENDIAN.iter()) {
+            less_than_modulus |= value_byte.ct_lt(modulus_byte) & !greater_than_modulus;
+            greater_than_modulus |= value_byte.ct_gt(modulus_byte) & !less_than_modulus;
+        }
+
+        match (
+            bool::from(less_than_modulus),
+            bool::from(greater_than_modulus),
+        ) {
+            (true, false) => {}
+            (false, _) => return Err(FieldError::ModulusOverflow),
+            (true, true) => unreachable!(),
+        }
+
+        // Switch from big endian to little endian.
```

Comment:
It would be helpful here to explain that VDAF and `fiat_crypto` expect opposite endianness, as you did in the `Encode` impl.